### ISSUE TITLE
Hibernate throwing exceptions when using MS SQL and `quarkus.hibernate-orm.database.generation=update`

### ISSUE THREAD

**Issue by wjglerum** [quarkusio/quarkus#13605]
**Describe the bug**
Just after upgrading to 1.10.2.Final we noticed that Hibernate can not longer update our tables. Even if they are in sync the underlying JDDBC driver will start throwing exceptions and Quarkus does not even start.

**Expected behavior**
Quarkus and Hibernate are able to introspect the tables on MS SQL and update them if needed.

**Actual behavior**
Quarkus fails to start
```
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< nl.wjglerum:quarkus-mssql >----------------------
[INFO] Building quarkus-mssql 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- quarkus-maven-plugin:1.10.2.Final:dev (default-cli) @ quarkus-mssql ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO] Nothing to compile - all classes are up to date
Listening for transport dt_socket at address: 5005
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2020-12-01 22:16:25,562 WARN  [org.hib.eng.jdb.spi.SqlExceptionHelper] (Quarkus Main Thread) SQL Error: 1038, SQLState: S0004
2020-12-01 22:16:25,574 ERROR [org.hib.eng.jdb.spi.SqlExceptionHelper] (Quarkus Main Thread) An object or column name is missing or empty. For SELECT INTO statements, verify each column has a name. For other statements, look for empty alias names. Aliases defined as "" or [] are not allowed. Change the alias to a valid name.
2020-12-01 22:16:25,604 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): com.microsoft.sqlserver.jdbc.SQLServerException: An object or column name is missing or empty. For SELECT INTO statements, verify each column has a name. For other statements, look for empty alias names. Aliases defined as "" or [] are not allowed. Change the alias to a valid name.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDatabaseError(SQLServerException.java:262)
	at com.microsoft.sqlserver.jdbc.TDSTokenHandler.onEOF(tdsparser.java:258)
	at com.microsoft.sqlserver.jdbc.TDSParser.parse(tdsparser.java:104)
	at com.microsoft.sqlserver.jdbc.TDSParser.parse(tdsparser.java:26)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection$1ConnectionCommand.doExecute(SQLServerConnection.java:2977)
	at com.microsoft.sqlserver.jdbc.TDSCommand.execute(IOBuffer.java:7194)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.executeCommand(SQLServerConnection.java:2935)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectionCommand(SQLServerConnection.java:2982)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.setCatalog(SQLServerConnection.java:3305)
	at com.microsoft.sqlserver.jdbc.SQLServerDatabaseMetaData.switchCatalogs(SQLServerDatabaseMetaData.java:381)
	at com.microsoft.sqlserver.jdbc.SQLServerDatabaseMetaData.getResultSetFromStoredProc(SQLServerDatabaseMetaData.java:335)
	at com.microsoft.sqlserver.jdbc.SQLServerDatabaseMetaData.getResultSetWithProvidedColumnNames(SQLServerDatabaseMetaData.java:357)
	at com.microsoft.sqlserver.jdbc.SQLServerDatabaseMetaData.getTables(SQLServerDatabaseMetaData.java:540)
	at org.hibernate.tool.schema.extract.internal.InformationExtractorJdbcDatabaseMetaDataImpl.getTables(InformationExtractorJdbcDatabaseMetaDataImpl.java:333)
	at org.hibernate.tool.schema.extract.internal.DatabaseInformationImpl.getTablesInformation(DatabaseInformationImpl.java:120)
	at org.hibernate.tool.schema.internal.GroupedSchemaMigratorImpl.performTablesMigration(GroupedSchemaMigratorImpl.java:65)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.performMigration(AbstractSchemaMigrator.java:207)
	at org.hibernate.tool.schema.internal.AbstractSchemaMigrator.doMigration(AbstractSchemaMigrator.java:114)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.performDatabaseAction(SchemaManagementToolCoordinator.java:184)
	at org.hibernate.tool.schema.spi.SchemaManagementToolCoordinator.process(SchemaManagementToolCoordinator.java:73)
	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:316)
	at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:73)
	at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:60)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
	at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:117)
	at io.quarkus.hibernate.orm.runtime.JPAConfig.startAll(JPAConfig.java:41)
	at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder.startAllPersistenceUnits(HibernateOrmRecorder.java:86)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits-2026547058.deploy_0(HibernateOrmProcessor$startPersistenceUnits-2026547058.zig:74)
	at io.quarkus.deployment.steps.HibernateOrmProcessor$startPersistenceUnits-2026547058.deploy(HibernateOrmProcessor$startPersistenceUnits-2026547058.zig:40)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:696)
	at io.quarkus.runtime.Application.start(Application.java:90)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:96)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:62)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:104)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:134)
	at java.base/java.lang.Thread.run(Thread.java:834)

Quarkus application exited with code 1
Press Enter to restart or Ctrl + C to quit
2020-12-01 22:16:25,606 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start hot replacement endpoint to recover from previous Quarkus startup failure
```

**To Reproduce**

Link to a small reproducer (preferably a Maven project if the issue is not Gradle-specific).

Managed to create simple reproducer, see https://github.com/wjglerum/quarkus-mssql

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.datasource.db-kind=mssql
quarkus.datasource.jdbc.url=jdbc:sqlserver://localhost
quarkus.datasource.username=sa
quarkus.datasource.password=yourStrong(!)Password
quarkus.http.access-log.enabled=true
quarkus.hibernate-orm.database.generation=update
#quarkus.hibernate-orm.database.generation=drop-and-create
#quarkus.hibernate-orm.database.generation=none
quarkus.hibernate-orm.log.sql=true
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: `Darwin Willems-MacBook-Pro.local 20.1.0 Darwin Kernel Version 20.1.0: Sat Oct 31 00:07:11 PDT 2020; root:xnu-7195.50.7~2/RELEASE_X86_64 x86_64`
 - Output of `java -version`: 
```
openjdk version "11.0.9.1" 2020-11-04
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.9.1+1)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.9.1+1, mixed mode)
```
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.10.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /Users/wjglerum/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 11.0.9.1, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
Default locale: en_NL, platform encoding: UTF-8
OS name: "mac os x", version: "10.16", arch: "x86_64", family: "mac"
```

**Additional context**
(Add any other context about the problem here.)
The app seems to start fine using `quarkus.hibernate-orm.database.generation=none`

**Comment by yrodiere** [comment]
Hey,

First off, sorry we didn't answer sooner. It seems your ticket slipped through the cracks.

I just tested your reproducer, and while I can reproduce the problem with Quarkus 1.10, it no longer happens on Quarkus 2.15.3 (though I had to change the JDBC URL to `jdbc:sqlserver://localhost;trustServerCertificate=true` to deal with self-signed certificates in the MS SQL container).

Thus it would seem your problem was addressed in an earlier version of Quarkus, probably by upgrading the mssql JDBC driver or Hibernate ORM dependencies in Quarkus.

Since this is no longer a problem, I'm going to close this issue. Feel free to reach out again if you still have issues, and I'll try to be more reactive this time :)
